.changeset/happy-lies-wink.md

```diff
@@ -0,0 +1,34 @@
+---
+'example-app': patch
+'@backstage/plugin-scaffolder': patch
+'@backstage/plugin-scaffolder-backend': patch
+---
+
+## `@backstage/scaffolder`
+
+- Adds a new field `EntityNamePicker` that can be used in scaffolder templates to accept and validate an entity name. This field is registered by default, and can be used in templates by setting the `ui:field` property to `EntityNamePicker`. If you've customized your scaffolder field extensions, you can include this one by adding it when registering the scaffolder route:
+
+```diff
+import {
+  ScaffolderFieldExtensions,
++   EntityNamePickerFieldExtension,
+} from '@backstage/plugin-scaffolder';
+
+  <Route path="/create" element={<ScaffolderPage />}>
+    <ScaffolderFieldExtensions>
+      {/* ...custom field extensions... */}
+
++       <EntityNamePickerFieldExtension />
+    </ScaffolderFieldExtensions>
+  </Route>;
+```
+
+- Adds a new generic field `TextValuePicker` to be used when writing custom field extensions that use a standard UI with custom validation. An example of doing this can be found in `packages/app/src/components/scaffolder/customScaffolderExtensions.tsx`.
+
+## `@backstage/scaffolder-backend
+
+- Updates the sample templates to use the new `EntityNamePicker` field, and cleans up some unused field configuration.
+
+## `example-app`
+
+- Adds an example of providing a custom field extension.
```

Comment:
We could split this into separate changeset files, looking at the divisions this one has using `##`. Or else, this whole thing will get copied over in all the `CHANGELOG` files. Also, `example-app` one is not needed, since we don't release the package!

Reply:
Makes sense! I'll add a second changeset. Perhaps it would be useful for `example-app` to be omitted from the list of packages in the `yarn changeset` prompt 🤔.

Reply:
Changeset updated - the other half ended up in #6726.

Reply:
Also - opened https://github.com/backstage/backstage/issues/6727 to look into omitting packages from the prompt in `yarn changeset`.